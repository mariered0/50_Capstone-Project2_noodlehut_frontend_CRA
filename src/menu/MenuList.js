import React, { useState, useEffect } from "react";
import CategoryCard from "./CategoryCard";
import NoodleHutApi from "../api/api";
import { Container, Typography } from "@mui/material";


const MenuList = () => {

    const [items, setItems] = useState([]);
    useEffect(() => {
        getMenu();
    }, []);


    async function getMenu() {
        const getData = await NoodleHutApi.getAllItems();

        //set only categoryNames
        const categories = getData.map(({categoryName}) => categoryName);
        const filtered = categories.filter(function (category, next) {return categories.indexOf(category) === next });
        setItems(filtered);
    }

    if (!items) return <h1>Loading...</h1>;

    return (
        <>
            <Container maxWidth='sm'>
            <Typography component="h1" variant="h2" align="center" color="textPrimary" gutterBottom>
              Menu
            </Typography>
            <Typography variant="h5" align="center" color="text.secondary" paragraph>
              Enjoy our authentic Thai food.
            </Typography>
           
            </Container>

                {items.map((categoryName, idx) => (
                    <CategoryCard 
                        key={idx}
                        category={categoryName}
                    />
                    ))}
          
        </>
    )
}

export default MenuList;