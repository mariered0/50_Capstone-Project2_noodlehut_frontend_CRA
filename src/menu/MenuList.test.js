import React from "react";
import { render } from '@testing-library/react';
import MenuList from "./MenuList";

it("renders without crashing", () => {
    render(
        <MenuList />
    );
});

it("matches snapshot", () => {
	const { asFragment } = render(
        <MenuList />
    );
	expect(asFragment()).toMatchSnapshot();
});