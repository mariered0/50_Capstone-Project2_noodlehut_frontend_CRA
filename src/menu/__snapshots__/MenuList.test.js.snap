// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  >
    <h1
      class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-10759ka-MuiTypography-root"
    >
      Menu
    </h1>
    <p
      class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiTypography-paragraph css-17rfbpd-MuiTypography-root"
    >
      Enjoy our authentic Thai food.
    </p>
  </div>
</DocumentFragment>
`;
